File: csharp/log-levels/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine) => logLine.Split(":")[1].Trim();

    public static string LogLevel(string logLine)
    {
        string errorLevel = logLine.Split(":")[0].Trim();
        return string.Join("", errorLevel.Split(':', '[', ']')).ToLower();
    }

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: solutions/cpp/log-levels/3/log_levels.cpp
#include <string>

namespace log_line {

    string message(string inputlog) {
        return inputlog.substr(inputlog.find(":"));
    }

    string log_level(string inputlog) {
        return inputlog;
    }

    string reformat(string inputlog) {
        return inputlog;
    }

} // namespace log_line
